File: Cpp/3637_trionic_array_i.cpp
// https://leetcode.cn/problems/trionic-array-i/

#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    bool isTrionic(vector<int>& nums) {
        int n = nums.size();
        int p = 0, q = 0;
        for (int i = 1; i < n; ++i) {
            if (nums[i] > nums[i-1]) p = i;
            else break;
        }
        for (int i = p+1; i < n; ++i) {
            if (nums[i] < nums[i-1]) q = i;
            else break;
        }
        for (int i = q+1; i < n; ++i) {
            if (nums[i] <= nums[i-1]) return false;
        }
        return 0 < p && p < q && q < n-1;
    }
};

int main() {
    Solution sol;
    vector<int> nums = {1,3,5,4,2,6};
    cout << sol.isTrionic(nums) << '\n';
    return 0;
}

File: Python3/3637_trionic_array_i.py
# https://leetcode.cn/problems/trionic-array-i/

class Solution:
    def isTrionic(self, nums: list[int]) -> bool:
        n = len(nums)
        p, q = 0, 0
        for i in range(1, n):
            if nums[i] > nums[i-1]:
                p = i
            else:
                break
        for i in range(p+1, n):
            if nums[i] < nums[i-1]:
                q = i
            else:
                break
        for i in range(q+1, n):
            if nums[i] <= nums[i-1]:
                return False
        return 0 < p and p < q and q < n-1